corehq/apps/userreports/tests/test_transforms.py

```diff
@@ -21,6 +21,34 @@ def testInvalidType(self):
             })
 
 
+class NumberFormatTransformTest(SimpleTestCase):
+
+    def setUp(self):
+        self.transform = TransformFactory.get_transform({
+            "type": "number_format",
+            "format_string": "{0:.0f}"
+        }).get_transform_function()
+
+    def test_int(self):
+        self.assertEqual('11', self.transform(11))
+
+    def test_decimal(self):
+        self.assertEqual('11', self.transform(11.23))
+
+    def test_strings(self):
+        tests = (
+            ('11', '11'),
+            ('11.23', '11'),
+            ('notanumber', 'notanumber'),
+            ('', ''),
+        )
```

Comment:
yeah i suppose it's a bit clunkier since it's outside the test class.
thanks for trying

On Mon, Jan 18, 2016 at 10:05 AM, Cory Zue <notifications@github.com> wrote:

> In corehq/apps/userreports/tests/test_transforms.py
> <https://github.com/dimagi/commcare-hq/pull/9918#discussion_r50008008>:
>
> > +            "format_string": "{0:.0f}"
> > +        }).get_transform_function()
> > +
> > +    def test_int(self):
> > +        self.assertEqual('11', self.transform(11))
> > +
> > +    def test_decimal(self):
> > +        self.assertEqual('11', self.transform(11.23))
> > +
> > +    def test_strings(self):
> > +        tests = (
> > +            ('11', '11'),
> > +            ('11.23', '11'),
> > +            ('notanumber', 'notanumber'),
> > +            ('', ''),
> > +        )
>
> not 100% sure i like this better, though looks like a useful utility and
> was worth the time it took to learn how to work with it. #9930
> <https://github.com/dimagi/commcare-hq/pull/9930>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/9918/files#r50008008>.
>